### DISCUSSION TITLE
Windows installation

### DISCUSSION THREAD

**Discussion by jeromeatneotek** [givanz/Vvveb#84] [General]
I have tried installing with both XAMPP and also using nginx with postgres. Both are able to get to the installation page and then setup the database, but as soon as I go to the dashboard I get SQL errors such as below:
`Internal server error!
(error 500)
FUNCTION myweb.JSON_ARRAYAGG does not exist
SELECT  pd.*,products.*, products.product_id as array_key

						 
				,(SELECT JSON_ARRAYAGG( JSON_OBJECT('id', pi.product_image_id, 'image', pi.image) ) 
				FROM product_image as pi WHERE pi.product_id = products.product_id GROUP BY pi.product_id) as images
				

		 
		FROM product AS products
		
			LEFT JOIN product_to_site p2s ON (products.product_id = p2s.product_id) 
			LEFT JOIN product_content pd ON (
				products.product_id = pd.product_id
		
						
					AND pd.language_id = 1
				

			)  

							
			
			
			WHERE p2s.site_id = 1

            		 		
			ORDER BY products.product_id DESC
		LIMIT 0, 4
 - SELECT  pd.*,products.*, products.product_id as array_key

						 
				,(SELECT JSON_ARRAYAGG( JSON_OBJECT('id', pi.product_image_id, 'image', pi.image) ) 
				FROM product_image as pi WHERE pi.product_id = products.product_id GROUP BY pi.product_id) as images
				

		 
		FROM product AS products
		
			LEFT JOIN product_to_site p2s ON (products.product_id = p2s.product_id) 
			LEFT JOIN product_content pd ON (
				products.product_id = pd.product_id
		
						
					AND pd.language_id = :language_id
				

			)  

							
			
			
			WHERE p2s.site_id = :site_id

            		 		
			ORDER BY products.product_id DESC
		LIMIT :start, :limit

C:\xampp\htdocs\myweb\system\db\mysqli.php on line 293
Code
throw new \Exception($message, $e->getCode()); // <==

		list($parameters, $types) = $this->paramsToQmark($sql, $params, $paramTypes);

		try {

			$stmt = self::$link->prepare($sql);

		} catch (\mysqli_sql_exception $e) {

			$message = $e->getMessage() . "\n" . $this->debugSql($origSql, $params, $paramTypes) . "\n - " . $origSql;

			throw new \Exception($message, $e->getCode());	 // <==

		}

		if ($stmt && ! empty($types)) {

			array_unshift($parameters, $types);

			//hack for php 7.x bind_param "expected to be a reference, value given" stupid warning

Trace
#0 C:\xampp\htdocs\myweb\storage\model\admin\productsql.mysqli.php(5536): Vvveb\System\Db\Mysqli->execute('SELECT  pd.*,pr...', Array, Array)
#1 C:\xampp\htdocs\myweb\admin\component\products.php(52): Vvveb\Sql\ProductSQL->getAll(Array)
#2 C:\xampp\htdocs\myweb\system\component\component.php(226): Vvveb\Component\Products->results()
#3 C:\xampp\htdocs\myweb\system\component\component.php(96): Vvveb\System\Component\Component->loadComponents()
#4 C:\xampp\htdocs\myweb\system\component\component.php(63): Vvveb\System\Component\Component->__construct(Object(Vvveb\System\Core\View), false, NULL)
#5 C:\xampp\htdocs\myweb\system\core\view.php(345): Vvveb\System\Component\Component::getInstance(Object(Vvveb\System\Core\View), false, NULL)
#6 C:\xampp\htdocs\myweb\system\core\response.php(102): Vvveb\System\Core\View->render()
#7 C:\xampp\htdocs\myweb\system\core\frontcontroller.php(251): Vvveb\System\Core\Response->output()
#8 C:\xampp\htdocs\myweb\system\core\frontcontroller.php(298): Vvveb\System\Core\FrontController::call('Vvveb\\Controlle...', 'index', 'C:\\xampp\\htdocs...')
#9 C:\xampp\htdocs\myweb\system\core\frontcontroller.php(364): Vvveb\System\Core\FrontController::redirect('Index', 'index')
#10 C:\xampp\htdocs\myweb\system\core\startup.php(355): Vvveb\System\Core\FrontController::dispatch()
#11 C:\xampp\htdocs\myweb\index.php(138): Vvveb\System\Core\start()
#12 C:\xampp\htdocs\myweb\admin\index.php(41): include('C:\\xampp\\htdocs...')
#13 {main}`

**Comment by givanz** [comment]
What mysql/mariadb version does your xampp installation have? 

You can view database version in `admin > tools > system info`

JSON_ARRAYAGG was added in MySQL version 5.7.22 and in MariaDB version 10.5.0

If you have an older version you need to upgrade

For pgsql issues please redownload latest.zip and update the files on your server, there were some bugs that have been fixed.

**Reply by jeromeatneotek** [reply]
Database driver & version
      mysqli | Client library version: mysqlnd 8.2.12 | Server version: 10.4.32-MariaDB

So I need to upgrade, the ability to find this out from system info was very helpful thanks

**Reply by jeromeatneotek** [reply]
I have reverted to using postgres with the new version it now works. However I tried turning on captcha and then restarted my browser so have been logged out and cannot login as the site doesn't have a captcha key loaded! Is there a way to turn off a plugin without logging in to admin?

**Reply by givanz** [reply]
If a plugin error prevents you to login to admin you can use safe mode to disable all plugins for the session then you can disable the plugin causing issues.

![safe-mode](https://github.com/givanz/Vvveb/assets/302738/d781959b-afd2-4d25-88cc-e1f33cf7f4be)


You can also manually disable any plugin by editing `config/plugins.php` and setting the plugin status to `inactive`

```php
    'captcha' => 
    array (
      'status' => 'inactive',
    ),
```

https://github.com/givanz/Vvveb/blob/master/config/plugins.php#L26
